drivers/hmis_csv_twenty_twenty_four/app/models/hmis_csv_twenty_twenty_four/exporter/client/overrides.rb

```diff
@@ -0,0 +1,151 @@
+    def process(row)
+      row = self.class.apply_overrides(row, export: @options[:export])
+
+      row
+    end
+
+    def self.apply_overrides(row, export:)
+      # Don't actually do this, identify duplicates and client cleanup should handle this, just use the destination client
+      # row = pick_best_source_client_data(row)
+      row = apply_length_limits(row)
+      row = apply_hash_status(row, export)
+      row = enforce_gender_none(row)
+      row = enforce_race_none(row)
+
+      [
+        :NameDataQuality,
+        :SSNDataQuality,
+        :DOBDataQuality,
+        :VeteranStatus,
+      ].each do |hud_field|
+        row = replace_blank(row, hud_field: hud_field, default_value: 99)
+      end
+
+      [
+        :Woman,
+        :Man,
+        :NonBinary,
+        :CulturallySpecific,
+        :Transgender,
+        :Questioning,
+        :DifferentIdentity,
+        :AmIndAKNative,
+        :Asian,
+        :BlackAfAmerican,
+        :HispanicLatinaeo,
+        :MidEastNAfrican,
+        :NativeHIPacific,
+        :White,
+      ].each do |hud_field|
+        row = replace_blank(row, hud_field: hud_field, default_value: 0)
+      end
+
+      row
+    end
+
+    def self.apply_hash_status(row, export)
+      return row unless export.hash_status == 4
+
+      row.FirstName = Digest::SHA256.hexdigest(Soundex.new(row.FirstName).soundex) if row.FirstName.present?
+      row.LastName = Digest::SHA256.hexdigest(Soundex.new(row.LastName).soundex) if row.LastName.present?
+      row.MiddleName = Digest::SHA256.hexdigest(Soundex.new(row.MiddleName).soundex) if row.MiddleName.present?
+
+      if row.SSN.present?
+        padded_ssn = row.SSN.rjust(9, 'x')
+        last_four =  padded_ssn.last(4)
+        digested_ssn = Digest::SHA256.hexdigest(padded_ssn)
+        row.SSN = "#{last_four}#{digested_ssn}"
+      end
+      row
+    end
+
+    def self.apply_length_limits(row)
+      row.FirstName = row.FirstName[0...50] if row.FirstName
+      row.MiddleName = row.MiddleName[0...50] if row.MiddleName
+      row.LastName = row.LastName[0...50] if row.LastName
+      row.NameSuffix = row.NameSuffix[0...50] if row.NameSuffix
+
+      row
+    end
+
+    def self.enforce_gender_none(row)
+      # GenderNone should be 99 if it was blank and all other gender columns are blank or 0
+      gender_columns = [
+        :Woman,
+        :Man,
+        :NonBinary,
+        :CulturallySpecific,
+        :Transgender,
+        :Questioning,
+        :DifferentIdentity,
+      ]
+      any_genders = gender_columns.map { |c| ! row[c].in?([nil, 0]) }.any?
+      row.GenderNone ||= 99 unless any_genders
+
+      row
+    end
+
+    def self.enforce_race_none(row)
+      # RaceNone should be 99 if it was blank and all other gender columns are blank or 0
+      race_columns = [
+        :AmIndAKNative,
+        :Asian,
+        :BlackAfAmerican,
+        :HispanicLatinaeo,
+        :MidEastNAfrican,
+        :NativeHIPacific,
+        :White,
+      ]
```

Comment:
```suggestion
      race_columns = HudUtility2024.race_id_to_field_name.values.uniq - [:RaceNone]
```

Reply:
I'll add `HudUtility2024.race_fields` to match gender_fields, but it doesn't exist currently.